projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/JobExecutionFilterSpec.java

```diff
@@ -41,6 +41,19 @@ public Predicate toPredicate(Root<DataJobExecution> root, CriteriaQuery<?> query
          if (CollectionUtils.isNotEmpty(filter.getStatusIn())) {
             predicates.add(root.get(DataJobExecution_.STATUS).in(filter.getStatusIn()));
          }
+
+         if (filter.getJobNameIn() != null && filter.getJobNameIn().size() > 0) {
+            predicates.add(root.get(DataJobExecution_.DATA_JOB).get(DataJob_.NAME).in(filter.getJobNameIn()));
```

Comment:
What is the reason that some filters use the criteria builder and some don't?

Reply:
I would imagine because builder has a convenience method greaterThanOrEqualTo/smallerThanOrEqualTo, whereas the root doesn't.